src/EnergyPlus/ChillerGasAbsorption.cc

```diff
@@ -173,24 +197,47 @@ namespace EnergyPlus::ChillerGasAbsorption {
 
     void GasAbsorberSpecs::getDesignCapacities(EnergyPlusData &state, const PlantLocation &calledFromLocation, Real64 &MaxLoad, Real64 &MinLoad, Real64 &OptLoad)
     {
-        // kind of a hacky way to find the location of this, but it's what plantloopequip was doing
-        int BranchInletNodeNum =
-            state.dataPlnt->PlantLoop(calledFromLocation.loopNum).LoopSide(calledFromLocation.loopSideNum).Branch(calledFromLocation.branchNum).NodeNumIn;
+        bool matchfound(false);
+
+        int branchTotalComp = state.dataPlnt->PlantLoop(calledFromLocation.loopNum)
+                                  .LoopSide(calledFromLocation.loopSideNum)
+                                  .Branch(calledFromLocation.branchNum)
+                                  .TotalComponents;
+
+        for (int iComp = 1; iComp <= branchTotalComp; iComp++) {
+            // kind of a hacky way to find the location of this, but it's what plantloopequip was doing
+            int compInletNodeNum = state.dataPlnt->PlantLoop(calledFromLocation.loopNum)
+                                       .LoopSide(calledFromLocation.loopSideNum)
+                                       .Branch(calledFromLocation.branchNum)
+                                       .Comp(iComp)
+                                       .NodeNumIn;
+
+            if (compInletNodeNum == this->ChillReturnNodeNum) { // Operate as chiller
+                MinLoad = this->NomCoolingCap * this->MinPartLoadRat;
+                MaxLoad = this->NomCoolingCap * this->MaxPartLoadRat;
+                OptLoad = this->NomCoolingCap * this->OptPartLoadRat;
+                matchfound = true;
+                break;
+            } else if (compInletNodeNum == this->HeatReturnNodeNum) { // Operate as heater
+                Real64 Sim_HeatCap = this->NomCoolingCap * this->NomHeatCoolRatio;
+                MinLoad = Sim_HeatCap * this->MinPartLoadRat;
+                MaxLoad = Sim_HeatCap * this->MaxPartLoadRat;
+                OptLoad = Sim_HeatCap * this->OptPartLoadRat;
+                matchfound = true;
+                break;
+            } else if (compInletNodeNum == this->CondReturnNodeNum) { // called from condenser loop
+                MinLoad = 0.0;
+                MaxLoad = 0.0;
+                OptLoad = 0.0;
+                matchfound = true;
+                break;
+            } else {
+                matchfound = false;
+            }
+        }
```

Comment:
If match found, update the loads values accordingly, break and return. Otherwise, continue to the next component check.